tough/src/editor/signed.rs

```diff
@@ -494,3 +504,187 @@ impl SignedRepository {
         Ok(())
     }
 }
+
+impl TargetsWalker for SignedDelegatedTargets {
+    fn targets(&self) -> HashMap<String, &Target> {
+        let mut targets_map = HashMap::new();
+        for targets in &self.roles {
+            targets_map.extend(targets.signed.signed.targets_map());
+        }
+        targets_map
+    }
+
+    fn consistent_snapshot(&self) -> bool {
+        self.consistent_snapshot
+    }
+}
+
+/// `PathExists` allows the user of our copy/link functions to specify what happens when the target
+/// is being written to a shared targets directory and the file already exists from another repo.
+#[derive(Debug, Deserialize, Clone, Copy)]
+#[serde(rename_all = "kebab-case")]
+pub enum PathExists {
+    /// Leave the existing file.
+    Skip,
+    /// Remove and replace the file; you might want this to update file metadata, for example.
+    Replace,
+    /// Stop writing targets and return an error.
+    Fail,
+}
+forward_from_str_to_serde!(PathExists);
+
+/// `TargetPath` represents an existing file at the path generated by `target_path`, if any, and
+/// the type of the file.  (Other file types will return an error instead.)  This can be used to
+/// determine whether you want to continue or fail.
+#[derive(Debug, Clone)]
+enum TargetPath {
+    /// No existing file found, we can create a new one at this path.
+    New { path: PathBuf },
+    /// Existing regular file found at this path.
+    File { path: PathBuf },
+    /// Existing symlink found at this path.
+    Symlink { path: PathBuf },
+}
+
+trait TargetsWalker {
```

Comment:
It looks like this unifies the logic for copying/linking targets for `SignedRepository` and `SignedDelegatedTargets`. Is that correct?  The logic is mostly the same?

Could we add a few lines of docstring to the trait and allude to the "default" implementations of `walk_targets()` and `target_path()`?